src/schema/query/activityLogs.query.ts

```diff
@@ -26,6 +27,32 @@ const FILTER_FIELDS: { name: string; type: string }[] = [
 
 /** Available sort fields */
 const SORT_FIELDS = [
+  {
+    name: 'username',
+    cursorId: (node: any) =>
+      (node.username || '') + ';;' + node.createdAt.getTime().toString(),
+    cursorFilter: (cursor: any, sortOrder: string) => {
+      const operator = sortOrder === 'asc' ? '$gt' : '$lt';
+      const value = decodeCursor(cursor).split(';;')[0];
+      const date = decodeCursor(cursor).split(';;')[1];
+      return {
+        $or: [
```

Comment:
Hi all, maybe someone with more experience on mongoose can help me out on this one
We have this property which can have the same value for multiple records, therefor on using pagination we'll have to store the value plus an additional unique property in order to have uniques cursors for each record in the ActivityLogs model.

What I did is use the value(or empty string) and the record createdAt value to build up the cursorId.
Then when receiving the cursor, to know from where I have to filter the values that I'll fetch(pagination) I just follow the next conditions:

If current username value is gt or lt, or if the value is eq to it and the createdAt value is gt. This way we make sure that even if there are still records in the next page with the same username value, are correctly paginated and fetched.

The problem is that this is not working, e.g. check the video

https://github.com/user-attachments/assets/3a241a7a-4cdf-4cd7-92f6-41b2e0647ce9


Any help is appreciated(i dont know if the filter condition order, or what is actually missing, or maybe my pagination logic for multiple records with the same value is not correct)
KR, Unai

cc @AntoineRelief , @Matthis-M-ReliefApps , @TaiKamilla , @MwanPygmay